### DISCUSSION TITLE
db.statement is blank in auto instrumentation trace for Cassandra

### DISCUSSION THREAD

**Discussion by AnilKumar356** [open-telemetry/opentelemetry-java-instrumentation#9820] [Q&A]
{"version": "0.2.0", "timestamp": "2023-11-08T05:27:09.345+01:00", "severity": "info", "service_id": "eric-bss-cpm-app", "metadata" : {"category": "CPM"}, "message": "i.o.j.i.i.l.Slf4jApplicationLoggerBridge$Slf4jApplicationLogger:60 'DB Query' : 50ea828b11ab29e84a3b95ccfe584863 45f17daa9e37ef56 CLIENT [tracer: io.opentelemetry.cassandra-3.0:1.31.0-alpha] AttributesMap{data={db.system=cassandra, db.statement=, net.sock.peer.addr=192.168.226.84, thread.id=1193, net.sock.peer.port=9042, net.sock.peer.name=eric-bss-cil-database-cd-cil1-rack1-0.eric-bss-cil-database-cd-cil1-rack1.eviupri.svc.cluster.local, thread.name=RMCA-Notification-Consumer}, capacity=128, totalAddedValues=7}"}

**Comment by laurit** [comment]
Does this happen for all or only for some statements? I guess this could happen when statement sanitizer is not able to parse the query, try disabling it https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/#db-statement-sanitization and let us know what the query is.

**Comment by AnilKumar356** [comment]
Yes it happening for all statements regardless of read or write operation. We also disable the sanitization bit still getting blank value. Query is basically ready and insert the data. Let us know what else I can do to debug this issue.

**Reply by laurit** [reply]
Could you provide a minimal sample application that reproduces the issue?

**Comment by AnilKumar356** [comment]
Actually our application don't use Cassandra client directly to persist the data. We have client api which is wrapper over Cassandra client. Client api have method like svaeEntity() or readEntity()
